### ISSUE TITLE
Adhearsion dies on invalid Asterisk response

### ISSUE THREAD

**Issue by bklang** [adhearsion/adhearsion#538]
It would be nice if this totally invalid Asterisk response didn't take down all the other calls in the system. As it is, this rendered the Adhearsion app unuseable.

```
[2015-02-03 13:53:56.988] TRACE Punchblock::Connection::Asterisk: [RECV] Event: VarSet^M
Privilege: dialplan,all^M
Timestamp: 1422989636.987881^M
Channel: SIP/304-0000d35d^M
Variable: PLAYBACKSTATUS^M
Value: SUCCESS^M
Uniqueid: 1422989628.59039^M
^M
Event: AGIExec^M
Privilege: agi,all^M
Timestamp: 1422989636.987950^M
SubEvent: End^M
Channel: SIP/304-0000d35d^M
CommandId: 653990461^M
Command: EXEC Playback "/usr/share/asterisk/sounds/en/confbridge-join"^M
ResultCode: 200^M
Result: Success^M
^M
Event: AsyncAGI^M
Privilege: agi,all^M
Timestamp: 1422989636.988115^M
SubEvent: Exec^M
Channel: SIP/304-0000d35d^M
CommandID: e0a71db7-2033-47dd-a737-13f82b0aae5a^M
Result: %24^M
^M

[2015-02-03 13:53:56.988] TRACE Punchblock::Connection::Asterisk: [RECV] #<RubyAMI::Event name="VarSet", headers={"Privilege"=>"dialplan,all", "Timestamp"=>"1422989636.987881", "Channel"=>"SIP/304-0000d35d", "Variable"=>"PLAYBACKSTATUS", "Value"=>"SUCCESS", "Uniqueid"=>"1422989628.59039"}, text_body=nil, events=[]>
[2015-02-03 13:53:56.989] TRACE Punchblock::Connection::Asterisk: [RECV] #<RubyAMI::Event name="AGIExec", headers={"Privilege"=>"agi,all", "Timestamp"=>"1422989636.987950", "SubEvent"=>"End", "Channel"=>"SIP/304-0000d35d", "CommandId"=>"653990461", "Command"=>"EXEC Playback \"/usr/share/asterisk/sounds/en/confbridge-join\"", "ResultCode"=>"200", "Result"=>"Success"}, text_body=nil, events=[]>
[2015-02-03 13:53:56.989] TRACE Punchblock::Connection::Asterisk: [RECV] #<RubyAMI::Event name="AsyncAGI", headers={"Privilege"=>"agi,all", "Timestamp"=>"1422989636.988115", "SubEvent"=>"Exec", "Channel"=>"SIP/304-0000d35d", "CommandID"=>"e0a71db7-2033-47dd-a737-13f82b0aae5a", "Result"=>"%24"}, text_body=nil, events=[]>
[2015-02-03 13:53:56.991] TRACE Punchblock::Connection::Asterisk: [RECV] Event: Hangup^M
Privilege: call,all^M
Timestamp: 1422989636.990796^M
Channel: SIP/lsi-0000d37f^M
Uniqueid: 1422989634.59069^M
CallerIDNum: 2405420185^M
CallerIDName: <unknown>^M
ConnectedLineNum: 2405420185^M
ConnectedLineName: <unknown>^M
AccountCode: ^M
Cause: 16^M
Cause-txt: Normal Clearing^M
^M
Event: OriginateResponse^M
Privilege: call,all^M
Timestamp: 1422989636.990889^M
ActionID: 1b5c13f6-2557-4076-8d0b-6f3f88457693^M
Response: Failure^M
Channel: SIP/lsi/<MASKED>^M
Context: adhearsion-redirect^M
Exten: 1^M
Reason: 1^M
Uniqueid: <null>^M
CallerIDNum: 2405420185^M
CallerIDName: <unknown>^M
^M

[2015-02-03 13:53:56.991] TRACE Punchblock::Connection::Asterisk: [RECV] #<RubyAMI::Event name="Hangup", headers={"Privilege"=>"call,all", "Timestamp"=>"1422989636.990796", "Channel"=>"SIP/lsi-0000d37f", "Uniqueid"=>"1422989634.59069", "CallerIDNum"=>"2405420185", "CallerIDName"=>"<unknown>", "ConnectedLineNum"=>"2405420185", "ConnectedLineName"=>"<unknown>", "AccountCode"=>"", "Cause"=>"16", "Cause-txt"=>"Normal Clearing"}, text_body=nil, events=[]>
[2015-02-03 13:53:56.992] TRACE Punchblock::Connection::Asterisk: [RECV] #<RubyAMI::Event name="OriginateResponse", headers={"Privilege"=>"call,all", "Timestamp"=>"1422989636.990889", "ActionID"=>"1b5c13f6-2557-4076-8d0b-6f3f88457693", "Response"=>"Failure", "Channel"=>"SIP/lsi/<MASKED>", "Context"=>"adhearsion-redirect", "Exten"=>"1", "Reason"=>"1", "Uniqueid"=>"<null>", "CallerIDNum"=>"2405420185", "CallerIDName"=>"<unknown>"}, text_body=nil, events=[]>
[2015-02-03 13:53:56.994] DEBUG ClientHelper: HTTP GET to https://nitro3.powerhrg.com//api/v1/call_status/customer_needs_message.json WITH PARAMS {:customer_id=>692745, :customer_type=>"appointment"} RESULT {"need_message":false}
[2015-02-03 13:53:57.005] DEBUG ClientHelper: HTTP POST to https://nitro3.powerhrg.com//api/v1/call_status/started_dialing WITH PARAMS {:customer_id=>38145746, :customer_type=>"inside_sales"} RESULT
[2015-02-03 13:53:57.005] ERROR Celluloid: Punchblock::Translator::Asterisk crashed!
ArgumentError: The result string (%24) did not match the required format ((?-mix:^(?<code>\d{3})( result=(?<result>-?\d*))? ?(?<data>\(?.*\)?)?$)).
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/ruby_ami-2.2.1/lib/ruby_ami/agi_result_parser.rb:13:in `initialize'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/punchblock-2.5.3/lib/punchblock/translator/asterisk/agi_command.rb:24:in `parse_result'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/punchblock-2.5.3/lib/punchblock/translator/asterisk/component/asterisk/agi_command.rb:34:in `success_reason'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/punchblock-2.5.3/lib/punchblock/translator/asterisk/component/asterisk/agi_command.rb:24:in `handle_ami_event'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/punchblock-2.5.3/lib/punchblock/translator/asterisk/call.rb:120:in `process_ami_event'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/punchblock-2.5.3/lib/punchblock/translator/asterisk.rb:209:in `ami_dispatch_to_or_create_call'
        org/jruby/RubyHash.java:1366:in `each_pair'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/punchblock-2.5.3/lib/punchblock/translator/asterisk.rb:207:in `ami_dispatch_to_or_create_call'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/punchblock-2.5.3/lib/punchblock/translator/asterisk.rb:90:in `handle_ami_event'
        org/jruby/RubyKernel.java:1952:in `public_send'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/celluloid-0.15.2/lib/celluloid/calls.rb:25:in `dispatch'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/celluloid-0.15.2/lib/celluloid/calls.rb:122:in `dispatch'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/celluloid-0.15.2/lib/celluloid/actor.rb:322:in `handle_message'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/celluloid-0.15.2/lib/celluloid/actor.rb:416:in `task'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/celluloid-0.15.2/lib/celluloid/tasks.rb:55:in `initialize'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/celluloid-0.15.2/lib/celluloid/tasks.rb:47:in `initialize'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/celluloid-0.15.2/lib/celluloid/tasks/task_fiber.rb:13:in `create'
[2015-02-03 13:53:57.006] ERROR Adhearsion::Initializer: <ArgumentError> The result string (%24) did not match the required format ((?-mix:^(?<code>\d{3})( result=(?<result>-?\d*))? ?(?<data>\(?.*\)?)?$)).
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/ruby_ami-2.2.1/lib/ruby_ami/agi_result_parser.rb:13:in `initialize'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/punchblock-2.5.3/lib/punchblock/translator/asterisk/agi_command.rb:24:in `parse_result'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/punchblock-2.5.3/lib/punchblock/translator/asterisk/component/asterisk/agi_command.rb:34:in `success_reason'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/punchblock-2.5.3/lib/punchblock/translator/asterisk/component/asterisk/agi_command.rb:24:in `handle_ami_event'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/punchblock-2.5.3/lib/punchblock/translator/asterisk/call.rb:120:in `process_ami_event'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/punchblock-2.5.3/lib/punchblock/translator/asterisk.rb:209:in `ami_dispatch_to_or_create_call'
        org/jruby/RubyHash.java:1366:in `each_pair'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/punchblock-2.5.3/lib/punchblock/translator/asterisk.rb:207:in `ami_dispatch_to_or_create_call'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/punchblock-2.5.3/lib/punchblock/translator/asterisk.rb:90:in `handle_ami_event'
        org/jruby/RubyKernel.java:1952:in `public_send'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/celluloid-0.15.2/lib/celluloid/calls.rb:25:in `dispatch'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/celluloid-0.15.2/lib/celluloid/calls.rb:122:in `dispatch'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/celluloid-0.15.2/lib/celluloid/actor.rb:322:in `handle_message'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/celluloid-0.15.2/lib/celluloid/actor.rb:416:in `task'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/celluloid-0.15.2/lib/celluloid/tasks.rb:55:in `initialize'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/celluloid-0.15.2/lib/celluloid/tasks.rb:47:in `initialize'
        /data/apps/talkbox/shared/bundle/jruby/1.9/gems/celluloid-0.15.2/lib/celluloid/tasks/task_fiber.rb:13:in `create'
[2015-02-03 13:53:57.008] WARN  Celluloid: Terminating task: type=:call, meta={:method_name=>:execute_command}, status=:receiving
[2015-02-03 13:53:57.010] WARN  Celluloid: Terminating task: type=:call, meta={:method_name=>:execute_command}, status=:receiving
[2015-02-03 13:53:57.011] WARN  Celluloid: Terminating task: type=:call, meta={:method_name=>:execute_command}, status=:receiving
[2015-02-03 13:53:57.015] WARN  Celluloid: Terminating task: type=:call, meta={:method_name=>:execute_command}, status=:receiving
[2015-02-03 13:53:57.020] WARN  Celluloid: Terminating task: type=:call, meta={:method_name=>:execute_command}, status=:receiving
[2015-02-03 13:53:57.022] WARN  Celluloid: Terminating task: type=:call, meta={:method_name=>:execute_command}, status=:receiving
[2015-02-03 13:53:57.024] WARN  Celluloid: Terminating task: type=:call, meta={:method_name=>:execute_command}, status=:receiving
[2015-02-03 13:53:57.026] WARN  Celluloid: Terminating task: type=:call, meta={:method_name=>:execute_command}, status=:receiving
```

**Comment by bklang** [comment]
More requested logs:

```
[2015-02-03 13:53:56.596] TRACE Punchblock::Connection::Asterisk: [SEND] Action: agi^M
ActionID: 8524a3e4-16f4-4777-9e58-6a72873d12e6^M
Channel: SIP/304-0000d35d^M
Command: EXEC Playback "/usr/share/asterisk/sounds/en/confbridge-join"^M
CommandID: e0a71db7-2033-47dd-a737-13f82b0aae5a^M

Event: AsyncAGI^M
Privilege: agi,all^M
Timestamp: 1422989636.988115^M
SubEvent: Exec^M
Channel: SIP/304-0000d35d^M
CommandID: e0a71db7-2033-47dd-a737-13f82b0aae5a^M
Result: %24^M

[2015-02-03 13:53:56.989] TRACE Punchblock::Connection::Asterisk: [RECV] #<RubyAMI::Event name="AsyncAGI", headers={"Privilege"=>"agi,all", "Timestamp"=>"1422989636.988115", "SubEvent"=>"Exec", "Channel"=>"SIP/304-0000d35d", "CommandID"=>"e0a71db7-2033-47dd-a737-13f82b0aae5a", "Result"=>"%24"}, text_body=nil, events=[]>
[2015-02-03 13:53:56.991] TRACE Punchblock::Connection::Asterisk: [RECV] Event: Hangup^M

```

**Comment by bklang** [comment]
Fix in adhearsion/punchblock#240
